Include/internal/pycore_dict.h

```diff
@@ -323,6 +323,8 @@ _PyInlineValuesSize(PyTypeObject *tp)
 int
 _PyDict_DetachFromObject(PyDictObject *dict, PyObject *obj);
 
+PyDictObject *_PyObject_materialize_managed_dict_lock_held(PyObject *);
```

Comment:
Let's name this like the other functions: `_PyObject_MaterializeManagedDict_LockHeld` and move the definition up next to `_PyObject_MaterializeManagedDict`.